src/opencensus/proto/agent/README.md

```diff
@@ -57,6 +57,58 @@ are established.
 expired on Agent side. Library needs to start a new connection with a unique identifier
 (MAY be different than the previous one).
 
+## Implementation details of Agent Server
+
+This section describes the in-process implementation details of OC-Agent.
+
+![agent-implementation](agent-implementation.png)
+
+The Agent consists of three main parts:
+
+1. The interceptors of different instrumentation libraries, such as OpenCensus, Zipkin,
+Istio Mixer, Prometheus client, etc. Interceptors act as the “frontend” or “gateway” of
+Agent. In addition, there MAY be one special receiver for receiving configuration updates
+from outside.
+2. The core Agent module. It acts as the “brain” or “dispatcher” of Agent.
+3. The exporters to different monitoring backends or collector services, such as
+Omnition Collector, Stackdriver Trace, Jaeger, Zipkin, etc.
+
+### Interceptors
+
+Each interceptor can be connected with multiple instrumentation libraries. The
+communication protocol between interceptors and libraries is the one we described in the
+proto files (for example trace_service.proto). When a library opens the connection with the
+corresponding interceptor, the first message it sends must have the `Node` identifier. The
+interceptor will then cache the `Node` for each library, and `Node` is not required for
+the subsequent messages from libraries.
+
+### Agent Core
+
+Most functionalities of Agent are in Agent Core. Agent Core's responsibilies include:
+
+1. Accept `SpanProto` from each interceptor. Note that the `SpanProto`s that are sent to
+Agent Core must have `Node` associated, so that Agent Core can differentiate and group
+`SpanProto`s by each `Node`.
+2. Store and batch `SpanProto`s.
+3. Add additional metadata/information to `SpanProto` or `Node`. For example, in a k8s
```

Comment:
This part is confusing, what is `Node` in this instance? Whose `Node` is it? Agent Core's or the interceptor?

It is the interceptor's node and to remove the ambiguity we can perhaps update this to:
```
3. Augment each SpanProto or Node sent from the interceptor.
For example, in a Kubernetes container, Agent Core can detect the namespace, pod id
and container name and add them to its record of Node from interceptor
```

Reply:
Sounds good, updated.